File: core/src/main/java/com/highpowerbear/hpbanalytics/enums/Currency.java
package com.highpowerbear.hpbanalytics.enums;

/**
 * Created by robertk on 11/18/2017.
 */
public enum Currency {
    EUR,
    USD,
    AUD,
    GBP,
    CHF,
    JPY,
    KRW,
    HKD,
    SGD;

    public static Currency findByValue(String value) {
        Currency result = null;
        for (Currency currency : values()) {
            if (currency.name().equalsIgnoreCase(value)) {
                result = currency;
                break;
            }
        }
        return result;
    }
}

File: core/src/main/java/com/highpowerbear/hpbanalytics/enums/Currency.kt
package com.highpowerbear.hpbanalytics.enums

/**
 * Created by robertk on 11/18/2017.
 */
enum class Currency {
    EUR, USD, AUD, GBP, CHF, JPY, KRW, HKD, SGD;

    companion object {
        fun findByValue(value: String): Currency? {
            var result: Currency? = null
            for (currency in values()) {
                if (currency.name.equals(value, ignoreCase = true)) {
                    result = currency
                    break
                }
            }
            return result
        }
    }
}